common/version.py

```diff
@@ -90,13 +95,17 @@ def get_release_version(current_version, last_released_version=None, incremental
     else:
         if current_version is None:
             return None
+        # If the version increment strategy is "calver", we actually want to release the "next"
+        # version (AKA today's date, rather than the date we last released).
+        if version_increment is not None and version_increment.strategy == "calver":
+            current_version = version_increment.increment(current_version)
```

Comment:
Second meat of this change - while calculating the release version, if the version increment strategy is "calver", run the version increment method to get today's calendar version for release.